### PR TITLE
[WIP][Feature] Model Generator

### PR DESCRIPTION
**Pull request by chrisseto**

### CODE REVIEW

File: share/schemas/loader.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    def to_django_field(self, constructor):
+        mod_name, _, name = self.type.rpartition('.')
+        mod = _import_module(mod_name)
+        options = self.options.copy()
+
+        if self.is_relation:
+            options['to'] = constructor.name_for(constructor.schema.types[options['to']])
+
+        if 'choices' in self.options:
+            options['choices'] = [(choice.lower().title(), choice.lower()) for choice in options['choices']]
```
Comment (aaxelb) [inline share/schemas/loader.py]: Should this also strip spaces out of the db value?

File: schemas/SHARE/v2/internal/schema.yaml [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        type: django.db.models.PositiveIntegerField
+- !python/share.schema.SchemaType
+  - name: CreativeWork
+  - description: null
+  - meta:
+      verbose_name_plural: Creative Works
+  - fields: !!omap
+    - type: !python/share.schema.SchemaField
+        choices:
+        - creative work
```
Comment (aaxelb) [inline schemas/SHARE/v2/internal/schema.yaml]: Might be out of scope for this PR, but should we use `unknown` for the "base" work type? Like in AgentWorkRelation.type above.

File: schemas/SHARE/v2/internal/schema.yaml [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        type: django.db.models.TextField
+    - withdrawn: !python/share.schema.SchemaField
+        type: django.db.models.NullBooleanField
+- !python/share.schema.SchemaType
+  - name: WorkRelation
+  - description: null
+  - fields: !!omap
+    - type: !python/share.schema.SchemaField
+        choices:
+        - work relation
```
Comment (aaxelb) [inline schemas/SHARE/v2/internal/schema.yaml]: Ditto, `unknown`?

File: schemas/SHARE/v2/internal/schema.yaml [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  - name: Agent
+  - description: "An Agent is an entity that has the power to act, e.g. an individual\
+      \ person or a group of people.\n\n    Agents make decisions and produce or contribute\
+      \ to the production of creative works."
+  - meta:
+      index_together: [type, name]
+  - fields: !!omap
+    - type: !python/share.schema.SchemaField
+        choices:
+        - Agent
```
Comment (aaxelb) [inline schemas/SHARE/v2/internal/schema.yaml]: Tritto, `unknown`?

File: schemas/SHARE/v2/internal/schema.yaml [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        related_name: tag_relations
+        to: CreativeWork
+        type: django.db.models.ForeignKey
+- !python/share.schema.SchemaType
+  - name: AgentRelation
+  - description: null
+  - fields: !!omap
+    - type: !python/share.schema.SchemaField
+        choices:
+        - agent relation
```
Comment (aaxelb) [inline schemas/SHARE/v2/internal/schema.yaml]: Quatto, `unknown`?

File: share/schemas/contructor.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            return self._models
+
+        self._models = collections.OrderedDict()
+        for typ in self._sorted_types:
+            model = typ.to_django_model(self)
+            self._models[typ.name] = model
+
+        return self._models
+
+    def meta_for(self, typ):
```
Comment (aaxelb) [inline share/schemas/contructor.py]: It'd be cleaner to use a simile

File: schemas/SHARE/v2/internal/schema.yaml (hunk 2 above)
Comment (chrisseto) [inline schemas/SHARE/v2/internal/schema.yaml]: Might be best? It would probably be better than just "creative work"

**Pull request closed without merging**